src/components/CanvasWrapper.tsx

```diff
@@ -86,9 +89,24 @@ export default function CanvasWrapper(inputProps: CanvasWrapperProps): ReactElem
 
   // CANVAS PROPERTIES /////////////////////////////////////////////////
 
-  useEffect(() => {
-    canv.setMissingFrameCallback(setShowMissingFileIcon);
-  }, [canv, setShowMissingFileIcon]);
+  const onFrameChangedCallback = useCallback(
+    (isMissing: boolean) => {
+      setShowMissingFileIcon(isMissing);
+      if (props.showAlert && isMissing) {
+        props.showAlert({
+          type: "warning",
+          message: "Warning: One or more frames failed to load.",
+          description:
+            "Check your network connection and access to the dataset path, or use the browser console to view details. Otherwise, contact the dataset creator as there may be missing files.",
+          showDoNotShowAgainCheckbox: true,
+          closable: true,
+        });
+      }
+    },
+    [props.showAlert, setShowMissingFileIcon, canv]
+  );
+
+  canv.setOnFrameChangeCallback(onFrameChangedCallback);
```

Comment:
I'm a little unsure of this diff. Previously this was being called inside a useEffect.  Is the above useCallback sort of replacing that? Could you still be using useEffect here?

Reply:
I probably didn't need to be using `useEffect` in the first place!

`useCallback` caches the definition of `onFrameChangedCallback` unless its dependencies change, which I'm using as a minor optimization.

`canv.setOnFrameChangeCallback` doesn't really incur any costs, it just causes `canv` to set an internal field, so there's no real concern with running it on each update. (I could also cache this through a `useMemo` but again, this should be fine.)